Classes/Backend/TceMain.php

```diff
@@ -80,6 +80,91 @@ public function processDatamap_postProcessFieldArray($status, $table, $id, &$fie
         }
     }
 
+    /**
+     * Command pre processing method
+     *
+     * Deletion of child records must be done before parent record is deleted,
+     * so this logic is here in the preProcess
+     *
+     * @param string $command The TCEmain operation status, fx. 'update'
+     * @param string $table The table TCEmain is currently processing
+     * @param string $id The records id (if any)
+     * @param mixed $value
+     * @param DataHandler $reference Reference to the parent object (TCEmain)
+     * @param array $pasteUpdate
+     * @return void
+     */
+    public function processCmdmap_preProcess($command, $table, $id, $value, &$reference, &$pasteUpdate){
+        $doProcessCommand = in_array($command,['delete']);
+        $doProcessCommand = $doProcessCommand && 'tt_content' == $table;
+
+
+        if ($doProcessCommand) {
+            // A copy, or localisation (which is also a copy) was made. Cascade copy operations for child records.
+            $gridChildRecords = [];
+            $resolver = $this->objectManager->get(ProviderResolver::class);
+            $originalRecord = $this->recordService->getSingle($table, '*', $id);
+            $recordCanBeGrid = !empty($originalRecord['pi_flexform']);
```

Comment:
This assumption is wrong - any record can be a grid, regardless whether it uses a flexform.